Automattic/wp-calypso
# Checkout: Applied coupon should be removed when unavailable

#### Changes proposed in this Pull Request

* This PR fixes the bug that doesn't show the coupon input when a coupon gets unavailable while it is applied to the cart. The wrong message is also fixed.
<img width="1085" alt="Checkout_—_WordPress_com" src="https://user-images.githubusercontent.com/212034/100134288-54bf6400-2ecb-11eb-8baf-5d09c27024b9.png">

#### Testing instructions



* Apply a coupon to a paid plan in the cart on the checkout page.
  <img width="1084" alt="Checkout_—_WordPress_com" src="https://user-images.githubusercontent.com/212034/100135256-bcc27a00-2ecc-11eb-9002-2e8fb1fc0d16.png">
* **IMPORTANT!** Remove the coupon data from the server-side or get it expired.
* Switch to a variant product.
  <img width="1084" alt="Checkout_—_WordPress_com" src="https://user-images.githubusercontent.com/212034/100135475-08752380-2ecd-11eb-9e64-47054fd3402a.png">
* The coupon should be removed from the cart.
  <img width="1084" alt="Checkout_—_WordPress_com" src="https://user-images.githubusercontent.com/212034/100135524-19be3000-2ecd-11eb-8471-b831551bbc3b.png">
